### DISCUSSION TITLE
Make transform optional for ImageRuns

### DISCUSSION THREAD

**Discussion by jwerre** [dolanmiu/docx#1456] [Ideas / Requests]
It can be difficult to the know the exact demotions of an image particularly if the image is a buffers or base64 encoded. When I try to add an `ImageRun` without a `transform.width` *and* `transform.height` the document is corrupt and won't open. It would be nice to be able to place an `ImageRun` without any transformations at all. It would also be nice to specify *either* a `width` or a `height` and allow the image to scale proportionally—similar to the way CSS handles it.

**Comment by dolanmiu** [comment]
Its not possible im afraid. Node.js doesn't actually know how big the photo is, it just has the raw image data. 

In a browser environment, the size is known, but for compatibility between node and browser, we cant have separate implementations. 

This is why the size has to be set.

**Reply by jwerre** [reply]
understandable

**Comment by dyrector** [comment]
I had the same situation. I have images that would always be 3" wide, but height would be variable. I used some JavaScript right before `const doc = new Document ({...})` Maybe this will help you.

My image is wrapped in a `<div id="whole_graphic">` tag in an HTML form. The set width of the `<div>` is 600px. My project is in Angular, so you might have to revise your code accordingly.

`const entryImg = document.getElementByID('whole_graphic'); // Get full image including border`
`let imgHeight = entryImg?.clientHeight;  // Get height of image in px`
`let graphicHeight = imgheight! / 200;  // Calculate the height in inches`

Then in the ImageRun, the transformation code is thus:

`transformation: {`
  `width: (3.0 * 96),  // Inches * DPI`
  `height: (graphicHeight * 96),  // Inches * DPI`

Of course, I could have made `width: 288`, but I wanted to preserve the logic in case the set width should change at some point.

Hope this helps.

**Reply by keogami** [reply]
Hey, how did you reach 96 DPI?
